packages/mgt-components/src/components/mgt-person/mgt-person.ts

```diff
@@ -1195,11 +1203,11 @@ export class MgtPerson extends MgtTemplatedComponent {
     }
 
     let initials = '';
-    if (person.givenName) {
-      initials += person.givenName[0].toUpperCase();
+    if ((person as Person).givenName) {
+      initials += (person as Person).givenName[0].toUpperCase();
     }
-    if (person.surname) {
-      initials += person.surname[0].toUpperCase();
+    if ((person as Person).surname) {
+      initials += (person as Person).surname[0].toUpperCase();
```

Comment:
This is good work @Mnickii! Maybe to refactor it a little, you can add a typeguard for the Contact and Person types. This should make these lines be re-writable to something like:

```typescript
    
  isPerson = (obj: IDynamicPerson): obj is Person => {
    return 'personType' in obj && obj?.personType?.class === 'Person';
  };

  isContact = (obj: IDynamicPerson): obj is Contact => {
    return 'initials' in obj;
  };

  // some other lines of code

    if (this.isContact(person)) {
      return person?.initials;
    }

    let initials = '';
    if (this.isPerson(person)) {
      const name = person?.givenName || person?.surname || '';
      initials += name[0];
    }
```